src/zjs_i2c.c

```diff
@@ -207,8 +209,8 @@ static jerry_value_t zjs_i2c_write(const jerry_value_t function_obj,
     else {
         ZJS_PRINT("no I2C device is ready yet\n");
     }
-
-    return ZJS_UNDEFINED;
+    return (jerry_value_t)reply;
```

Comment:
And no the return number is either 0 for success or negative for error, its not the number of bytes.